### ISSUE TITLE
Crash on arrival

### ISSUE THREAD

**Issue by ianthetechie** [stadiamaps/ferrostar#126]
This manifests on iOS today with the arrival view on iOS, but I'm pretty sure the root cause is the core. Here's a sample backtrace.

```
Last Exception Backtrace:
0   CoreFoundation                	       0x19c428f20 __exceptionPreprocess + 164
1   libobjc.A.dylib               	       0x1942da018 objc_exception_throw + 59
2   Foundation                    	       0x19b92f868 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 187
3   Foundation                    	       0x19b3af308 -[NSDateComponentsFormatter stringFromTimeInterval:] + 255
4   iOS Demo                      	       0x100424270 closure #1 in ArrivalView.body.getter + 648
5   SwiftUI                       	       0x1a0305324 0x1a03052bd + 103
6   iOS Demo                      	       0x100422cac ArrivalView.body.getter + 616
7   iOS Demo                      	       0x100429280 protocol witness for View.body.getter in conformance ArrivalView + 12
```

TL;DR, there's some value being passed to the `DateComponentsFormatter` which is causing a crash. I tried several "obvious" edge cases like zero, negative zero (the value in question is a `Double`), and -1. Then I remembered that there are plenty more, because doubles... like `NaN`, infinity, and subnormals.

I believe that something in `algorithms.rs` (the `calculate_trip_progress` function) is responsible. I have observed similar issues with doing these sorts of calcs before; sometime even traceable back to the geo rust calculations, which have on occasion not been fully implemented or sound against edge cases (any sort of IEEE-754 division is inherently unsafe.)

**Comment by Archdoog** [comment]
@ianthetechie Closed by #127?

**Comment by ianthetechie** [comment]
Ahh yes, that should have fixed this.
